install/installer/cluster_dev.py

```diff
@@ -110,26 +118,55 @@ def validate(self, document=None, interactive=True):
             )  # Move cursor to end
 
 
-def ask_user(question_name, non_interactive_value=None):
+@typechecked
+class AWSUserNameValidator(Validator):  # pylint: disable=too-few-public-methods
+    """Validate user input"""
+
+    def validate(self, document=None, interactive=True):
+        """
+        Validate user input string to AWS username restrictions
+
+        Raises:
+            ValidationError: If intput string not match regex.
+                Show error message for user and get him change fix error
+        """
+        username = self
+        if interactive:
+            username = document.text
+
+        error_message = 'Please enter a valid AWS username. ' + \
+            'It should have 1-64 only latin letters, ' + \
+            'numbers and symbols: _+=,.@-'
+
+        okay = regex.match('^[A-Za-z0-9_+=,.@-]{1,64}$', username)
+
+        if not okay:
+            if not interactive:
+                _exit_(error_message)
+
+            raise ValidationError(
+                message=error_message,
+                cursor_position=len(username),
+            )  # Move cursor to end
```

Comment:
Pro tip: employ a guard-expression style check in order to dedent the whole block that follows it. To do so, just invert the first if-clause and interrupt the flow early:
```suggestion
        if okay:
            return

        if not interactive:
            _exit_(error_message)

        raise ValidationError(
            message=error_message,
            cursor_position=len(username),
        )  # Move cursor to end
```

Reply:
Done